src/openrct2/management/NewsItem.cpp

```diff
@@ -295,18 +309,17 @@ std::optional<CoordsXYZ> news_item_get_subject_location(int32_t type, int32_t su
     return subjectLoc;
 }
 
-static NewsItem* news_item_first_open_queue_slot()
+NewsItem* NewsItemQueue::FirstOpenSlot()
```

Comment:
The name suggests const, but the implementation actually mutates. Can you come up with better name for this method?

Reply:
Changed to FirstOpenOrCreateSlot

Reply:
That's an odd name, it reads as if "open" was a verb in this context. I'd suggest FirstOpenOrNewSlot

Reply:
Done 😄